sympy/physics/mechanics/kane.py

```diff
@@ -63,9 +76,39 @@ class KanesMethod(_Methods):
     are given in the explicit form by default. In other words, the kinematic
     mass matrix is $\mathbf{k_{k\dot{q}}} = \mathbf{I}$.
     In order to get the implicit form of those matrices/vectors, you can set the
-    ``explicit_kinematics`` attribute to ``False``. So $\mathbf{k_{k\dot{q}}}$ is not
-    necessarily an identity matrix. This can provide more compact equations for
-    non-simple kinematics (see #22626).
+    ``explicit_kinematics`` attribute to ``False``. So $\mathbf{k_{k\dot{q}}}$
+    is not necessarily an identity matrix. This can provide more compact
+    equations for non-simple kinematics (see #22626).
+
+    Two linear solvers can be supplied to ``KanesMethod``: one for solving the
+    kinematic differential equations and one to solve the velocity constraints.
+    Both of these sets of equations form a linear system ``Ax = b``, which has
+    to be solved in order to obtain the equations of motion.
+
+    The implemented solvers are:
+        - ``'lu'``: LU solver which utilizes ``Matrix.LUsolve`` (default)
+        - ``'numerics'``: numerical solver which utilizes
+          ``codegen.matrix_nodes.MatrixSolve``. Sets ``.use_block_matrices`` to
+          ``True``.
+        - ``callable``: custom solver supplied by the user, which takes the
+          ``A`` matrix as the first argument and ``b`` matrix as the second.
```

Comment:
Move these options into the parameter explanation.

Reply:
Isn't there a location where we can just state this once. Otherwise you'll be having a copy of this part on a lot of different locations: twice in the `KanesMethod` docstring, once in the `rhs` and also in `Linearize`.

Reply:
In interactive us a user will type either `help(KanesMethod)` when initializing a class or `help(kane_object.method())` when using a method. The necessary info for doing either should be immediately visible. If you don't want redundant info in the source code you can always programmatically construct the `__doc__` attribute for the methods that share common text.